File: 9/golang/main.go
// A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
//
// a^2 + b^2 = c^2
// For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
//
// There exists exactly one Pythagorean triplet for which a + b + c = 1000.
// Find the product abc.

package main

import (
    "fmt"
    "math"
)

func eq(a, b int) float64 {
    af, bf := float64(a), float64(b)
    root := math.Sqrt(af*af + bf*bf)
    return ( af * (af + root) ) + ( af * bf ) + (bf * (bf + root) )
}

func solveForABC() (int, int, int) {
    for a := 1; a < 998; a++ {
        for b := 1; b < 998; b++ {
            if eq(a, b) == float64(500000) {
                return a, b, 1000-a-b
            }
        }
    }
    return 0, 0, 0
}

func main() {
    a, b, c := solveForABC()
    result := a * b * c
    fmt.Println(result)
}

File: 9/python/main.py
# A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
#
# a^2 + b^2 = c^2
# For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
#
# There exists exactly one Pythagorean triplet for which a + b + c = 1000.
# Find the product abc.

import math

def eq(a, b):
    root = math.sqrt(a*a + b*b)
    return ( a * (a + root) ) + ( a * b ) + (b * (b + root) )

def solve_for_a_b_c():
    for a in range(1, 998):
        for b in range(1, 998):
            if eq(a, b) == 500000:
                return a, b, 1000 - a - b

a, b, c = solve_for_a_b_c()
print(a*b*c)
